Source/HyprMXAdapterInterstitialAd.swift

```diff
@@ -18,21 +18,28 @@ final class HyprMXAdapterInterstitialAd: HyprMXAdapterAd, PartnerAd {
     /// - parameter completion: Closure to be performed once the ad has been loaded.
     func load(with viewController: UIViewController?, completion: @escaping (Result<PartnerEventDetails, Error>) -> Void) {
         log(.loadStarted)
-        loadCompletion = completion
 
         // HyprMX only supports interaction from the Main Thread. Unlike banner ads, Chartboost Mediation SDK
         // does not load interstitials on the main thread so we need to wrap this ourselves.
         DispatchQueue.main.async { [self] in
             // Construct a partner ad to be persisted for subsequent ad operations.
             if let ad = HyprMX.getPlacement(self.request.partnerPlacement) {
                 self.ad = ad
-                ad.placementDelegate = self
-                ad.loadAd()
+                ad.expiredDelegate = self
+                ad.loadAd { success in
+                    if success {
+                        self.log(.loadSucceeded)
+                        completion(.success([:]))
+                    } else {
+                        let loadError = self.error(.loadFailureNoFill)
```

Comment:
Since no error details are provided by the partner, there's no way to know if the load failed due to no fill or due to some other error right?
```suggestion
                        let loadError = self.error(.loadFailureUnknown)
```

Reply:
Yeah, they seem to provide less error data now.